policy/nomad/source.go

```diff
@@ -164,6 +151,16 @@ func (s *Source) MonitorIDs(ctx context.Context, req policy.MonitorIDsReq) {
 
 		// Send new policy IDs in the channel.
 		req.ResultCh <- policy.IDMessage{IDs: policyIDs, Source: s.Name()}
+
+		select {
+		case <-ctx.Done():
+			s.log.Trace("stopping ID subscription")
+			return
+		case <-s.reloadCh:
+			s.log.Trace("reloading policies")
+			continue
+		case <-ticker.C:
+		}
```

Comment:
hm, is this new ticker here to account for cases where the blocking query unblocks very rapidly?  wouldn't this add an extra half-second to all calls, even those that properly block waiting for changes?